results/cmd/api/main_test.go

```diff
@@ -555,7 +555,10 @@ func TestListTaskRuns(t *testing.T) {
 			if tc.expectStatus != status.Code(err) {
 				t.Fatalf("failed on test %v: %v", tc.name, err)
 			}
-			gotList := res.GetItems()
+			var gotList []*pb.TaskRun
+			for _, taskrunResult := range res.GetItems() {
+				gotList = append(gotList, taskrunResult.GetTaskRun())
+			}
```

Comment:
Only looking at the taskruns means that we are ignoring the extra data being returned by TaskRunResults. We should modify the expected results to include the TaskRunResults so we can verify that it's being set properly.

We'll need a way to optionally make ID values deterministic for testing (easiest thing to do is likely just making the ID generator func configurable by the package, and setting it to something deterministic in tests).

Reply:
done